src/amo/components/LandingPage.js

```diff
@@ -108,9 +115,22 @@ export class LandingPageBase extends React.Component {
   }
 }
 
-export function mapStateToProps(state, ownProps) {
+export function mapStateToProps(
+  state, ownProps, _apiAddonType = apiAddonType, _log = log
+) {
+  let addonType;
+  try {
+    addonType = _apiAddonType(ownProps.params.visibleAddonType);
+  } catch (err) {
+    if (err.message.match('not found in API_ADDON_TYPES_MAPPING')) {
```

Comment:
I don't think we have a lot of (any?) custom error types yet but I bet that wouldn't be hard. I like it 👍